app/Helpers/shortcode.php

```diff
@@ -58,11 +59,29 @@ function url2shortcode(string $url): string
     return $link;
 }
 
+function convertUserUrlsToShortcodes(string $input): string
+{
+    // Find all instances of user URLs in the input.
+    preg_match_all("~https?://(?:[\w\-]+\.)?retroachievements\.org/user/([\w\-]+)(/?(?![\w/?]))~i", $input, $matches);
```

Comment:
localhost user link seems to be handled by the preview functionality, but not this code.
![image](https://github.com/user-attachments/assets/dd83c885-d22b-4372-b5e2-1fee39aabdd9)

Reply:
Swapping the execution order of `Shortcode::convertUserShortcodesToUseIds()` and `normalize_shortcodes()` seems to resolve this issue. Updated in latest.

Reply:
Seems to be working in forums posts, but not messages.
![image](https://github.com/user-attachments/assets/c1552e42-bb4c-47d1-9110-76812a98e4cb)

Reply:
Should be fixed in latest. Looks like this issue also happens in prod?

Reply:
Verified the URL->[user] functionality is not present in master. The delete user is handled correctly there.

Everything seems to be working correctly on the branch now.